pkg/autoscaler/autoscaler.go

```diff
@@ -877,4 +895,130 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
+			specReport := CurrentSpecReport()
+			if specReport.Failed() {
+				Expect(gatherer.WithSpecReport(specReport).GatherAll()).To(Succeed(), "Failed to gather spec report")
+			}
+
+			// explicitly delete the ClusterAutoscaler
+			// this is needed due to the autoscaler tests requiring singleton
+			// deployments of the ClusterAutoscaler.
+			By("Waiting for ClusterAutoscaler to delete.")
+			caName := clusterAutoscaler.GetName()
+			Expect(deleteObject(caName, cleanupObjects[caName])).Should(Succeed(), "Failed to delete ClusterAutoscaler")
+			delete(cleanupObjects, caName)
+			Eventually(func() (bool, error) {
+				_, err := framework.GetClusterAutoscaler(client, caName)
+				if apierrors.IsNotFound(err) {
+					return true, nil
+				}
+				// Return the error so that failures print additional errors
+				return false, err
+			}, framework.WaitMedium, pollingInterval).Should(BeTrue(), "Failed to cleanup Cluster Autoscaler before timeout")
+		})
+
+		// Machines required for test: 4
+		// Reason: This test starts with 2 machinesets, each with 1 replica to avoid scaling from zero.
+		// Then it autoscales both machinesets to 2 replicas.
+		// Does not start with replicas=0 machineset to avoid scaling from 0.
+		// OCP-73446 - Cluster autoscaler support priority expander option
+		// author: zhsun@redhat.com
+		It("high priority machineset should be scaled up first [Slow]", func() {
+			By("Creating 2 MachineSets each with 1 replica")
+			var transientMachineSets [2]*machinev1.MachineSet
+			targetedNodeLabel := fmt.Sprintf("%v-priority-expander", autoscalerWorkerNodeRoleLabel)
+			for i := range transientMachineSets {
+				machineSetParams := framework.BuildMachineSetParams(ctx, client, 1)
+				machineSetParams.Labels[targetedNodeLabel] = ""
+				machineSet, err := framework.CreateMachineSet(client, machineSetParams)
+				Expect(err).ToNot(HaveOccurred(), "Failed to create MachineSet %d of %d", i, len(transientMachineSets))
+				cleanupObjects[machineSet.GetName()] = machineSet
+				transientMachineSets[i] = machineSet
+			}
+
+			By("Waiting for all Machines in MachineSets to enter Running phase")
+			framework.WaitForMachineSet(ctx, client, transientMachineSets[0].GetName())
+			framework.WaitForMachineSet(ctx, client, transientMachineSets[1].GetName())
+
+			maxMachineSetReplicas := int32(2)
+			for _, machineSet := range transientMachineSets {
+				By(fmt.Sprintf("Creating a MachineAutoscaler backed by MachineSet %s - min: 1, max: %d",
+					machineSet.GetName(), maxMachineSetReplicas))
+				asr := machineAutoscalerResource(machineSet, 1, maxMachineSetReplicas)
+				Expect(client.Create(ctx, asr)).Should(Succeed(), "Failed to create MachineAutoscaler with min 1/max %d replicas", maxMachineSetReplicas)
+				cleanupObjects[asr.GetName()] = asr
+			}
+
+			By("Create cluster-autoscaler-priority-expander")
+			priorities := "10: \n  - .*" + transientMachineSets[0].GetName() + ".*\n20:\n  - .*" + transientMachineSets[1].GetName() + ".*"
```

Comment:
This would be easier to understand if it were a multi-line string with a `fmt.Sprintf`, WDYT?